src/dearpygui_commands.h

```diff
@@ -2516,6 +2687,9 @@ destroy_context(PyObject* self, PyObject* args, PyObject* kwargs)
 			GContext->started = false;  // return to false after
 			});
 
+		if (GContext->viewport != nullptr)
+			mvCleanupViewport(*GContext->viewport);
```

Comment:
`mvCleanupViewport`: Calling it here creates a race condition where `onCloseCallback` competes with the rest of `destroy_context` for DPG and ImGui objects. This is a known issue with `destroy_context` and I have a fix for it; however, right now let's not aggravate it by adding `mvCleanupViewport` in the mix.

It should be much safer to call it after `GContext->future.get()`. Let's just move the call to occur right before `delete GContext->viewport` (below),  in the same `if` block.

Reply:
I see. `mvCleanupViewport` calls platform-specific cleanup in ImGui (which is not called at all in the old version), so it needs to go before `ImGui::DestroyContext()`. Let's leave it where it is in your version then. I'm going to fix that in future. Unfortunately the fix is not trivial and therefore I'd prefer not to include it directly into this update; I'll open a separate PR later.